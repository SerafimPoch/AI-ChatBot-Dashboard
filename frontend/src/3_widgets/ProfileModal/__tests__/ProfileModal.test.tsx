import React from "react";
import { render } from "@testing-library/react";
import {
  ProfileInfoWrapper,
  ProfileName,
  ProfileTag,
  ProfileInfoOptionList,
  ProfileThemeWrapper,
  ProfileLogoutWrapper,
  ProfileItemWrapper,
  ProfileItemText,
} from "../styles";

describe("ProfileModal page components", () => {
  it("ProfileInfoWrapper renders correctly", () => {
    const { asFragment } = render(<ProfileInfoWrapper />);
    expect(asFragment()).toMatchSnapshot();
  });
  it("ProfileName renders correctly", () => {
    const { asFragment } = render(<ProfileName />);
    expect(asFragment()).toMatchSnapshot();
  });
  it("ProfileTag renders correctly", () => {
    const { asFragment } = render(<ProfileTag />);
    expect(asFragment()).toMatchSnapshot();
  });
  it("ProfileInfoOptionList renders correctly", () => {
    const { asFragment } = render(<ProfileInfoOptionList />);
    expect(asFragment()).toMatchSnapshot();
  });
  it("ProfileThemeWrapper renders correctly", () => {
    const { asFragment } = render(<ProfileThemeWrapper />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("ProfileLogoutWrapper renders correctly", () => {
    const { asFragment } = render(<ProfileLogoutWrapper />);
    expect(asFragment()).toMatchSnapshot();
  });
  it("ProfileItemWrapper renders correctly", () => {
    const { asFragment } = render(<ProfileItemWrapper />);
    expect(asFragment()).toMatchSnapshot();
  });
  it("ProfileItemText renders correctly", () => {
    const { asFragment } = render(<ProfileItemText />);
    expect(asFragment()).toMatchSnapshot();
  });
});
