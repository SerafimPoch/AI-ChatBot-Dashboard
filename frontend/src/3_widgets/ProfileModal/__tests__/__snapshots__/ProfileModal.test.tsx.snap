// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileModal page components ProfileInfoOptionList renders correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  margin: 0 -28px;
  padding: 8px 12px;
  border-bottom: 1px solid #f2f2f2;
}

<div
    class="emotion-0"
  />
</DocumentFragment>
`;

exports[`ProfileModal page components ProfileInfoWrapper renders correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  padding: 0 24px 24px 24px;
  margin: 0 -28px;
  border-bottom: 1px solid #f2f2f2;
}

<div
    class="emotion-0"
  />
</DocumentFragment>
`;

exports[`ProfileModal page components ProfileItemText renders correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  font-size: 16px;
  font-weight: 600;
  line-height: 24px;
  font-family: var(--font-sf-pro-rounded);
  color: var(--text-secondary);
}

<p
    class="emotion-0"
  />
</DocumentFragment>
`;

exports[`ProfileModal page components ProfileItemWrapper renders correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 12px;
  padding: 12px 16px;
  margin-bottom: 4px;
  cursor: pointer;
}

<div
    class="emotion-0"
  />
</DocumentFragment>
`;

exports[`ProfileModal page components ProfileLogoutWrapper renders correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  margin: 0 -28px;
  padding: 8px 12px;
  border-bottom: 1px solid #f2f2f2;
  border: none;
}

<div
    class="emotion-0"
  />
</DocumentFragment>
`;

exports[`ProfileModal page components ProfileName renders correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  font-size: 18px;
  font-weight: 600;
  line-height: 24px;
  font-family: var(--font-sf-pro-rounded);
  color: undefined;
}

<p
    class="emotion-0"
  />
</DocumentFragment>
`;

exports[`ProfileModal page components ProfileTag renders correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  font-size: 14px;
  font-weight: 500;
  line-height: 20px;
  font-family: var(--font-sf-pro-rounded);
  color: undefined;
}

<p
    class="emotion-0"
  />
</DocumentFragment>
`;

exports[`ProfileModal page components ProfileThemeWrapper renders correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  margin: 0 -28px;
  padding: 20px 12px;
  border-bottom: 1px solid #f2f2f2;
}

<div
    class="emotion-0"
  />
</DocumentFragment>
`;
